parser/parser.go

```diff
@@ -126,10 +125,10 @@ func (p *Parser) parseNode() (n ast.Node, eof bool, err error) {
 	// parsers should not leave tokens not processed
 	// in the lookahead buffer.
 	if len(p.lookahead) != 0 {
-		panic(fmt.Sprintf(
+		return nil, false, fmt.Errorf(
```

Comment:
why not panic? (just asking for the motivation)

Reply:
the error the tests were a lot harder to understand with panic, after using a error return the error was much easier to track. But if on other scenarios it will be more clear to panic we can change back again.